pkg/localhelm/helm3.go

```diff
@@ -169,114 +144,216 @@ func Deploy(st *state.State) error {
 	if err != nil {
-		return fail.Runtime(err, "upgrading helm release %q from chart %q", rh.Chart, rh.ReleaseName)
+		return fail.Runtime(err, "upgrading helm release %q from chart %q", release.Chart, release.ReleaseName)
 	}
 
 	secretObjectKey := ctrlruntimeclient.ObjectKey{
 		Name:      makeKey(rel.Name, rel.Version),
-		Namespace: rh.Namespace,
+		Namespace: release.Namespace,
 	}
 
 	return addReleaseSecretLabels(ctx, secretObjectKey, dynclient)
 }
 
+func newHelmInstallClient(cfg *helmaction.Configuration, release kubeoneapi.HelmRelease) *helmaction.Install {
+	helmInstall := helmaction.NewInstall(cfg)
+	helmInstall.DependencyUpdate = true
+	helmInstall.CreateNamespace = true
+	helmInstall.IncludeCRDs = true
+	helmInstall.Namespace = release.Namespace
+	helmInstall.ReleaseName = release.ReleaseName
+	helmInstall.RepoURL = release.RepoURL
+	helmInstall.Version = release.Version
+
+	return helmInstall
+}
+
 func installRelease(
 	ctx context.Context,
 	cfg *helmaction.Configuration,
-	rh kubeoneapi.HelmRelease,
+	release kubeoneapi.HelmRelease,
 	helmSettings *helmcli.EnvSettings,
 	providers getter.Providers,
 	dynclient ctrlruntimeclient.Client,
 	vals map[string]interface{},
+	logger logrus.FieldLogger,
 ) error {
-	helmInstall := helmaction.NewInstall(cfg)
-	helmInstall.DependencyUpdate = true
-	helmInstall.CreateNamespace = true
-	helmInstall.Namespace = rh.Namespace
-	helmInstall.ReleaseName = rh.ReleaseName
-	helmInstall.RepoURL = rh.RepoURL
-	helmInstall.Version = rh.Version
-
-	chartRequested, err := getChart(rh.Chart, helmInstall.ChartPathOptions, helmSettings, providers)
+	logger.Infof("Deploying helm chart %s as release %s", release.Chart, release.ReleaseName)
+	helmInstall := newHelmInstallClient(cfg, release)
+	rel, err := runInstallRelease(ctx, release, helmInstall, helmSettings, providers, vals)
 	if err != nil {
 		return err
 	}
 
-	rel, err := helmInstall.RunWithContext(ctx, chartRequested, vals)
-	if err != nil {
-		return fail.Runtime(err, "installing helm release %q from chart %q", rh.Chart, rh.ReleaseName)
-	}
-
 	secretObjectKey := ctrlruntimeclient.ObjectKey{
 		Name:      makeKey(rel.Name, rel.Version),
-		Namespace: rh.Namespace,
+		Namespace: release.Namespace,
 	}
 
 	return addReleaseSecretLabels(ctx, secretObjectKey, dynclient)
 }
 
+func runInstallRelease(
+	ctx context.Context,
+	release kubeoneapi.HelmRelease,
+	client *helmaction.Install,
+	helmSettings *helmcli.EnvSettings,
+	providers getter.Providers,
+	vals map[string]interface{},
+) (*helmrelease.Release, error) {
+	chartRequested, err := getChart(release.Chart, client.ChartPathOptions, helmSettings, providers)
+	if err != nil {
+		return nil, err
+	}
+
+	rel, err := client.RunWithContext(ctx, chartRequested, vals)
+	if err != nil {
+		return nil, fail.Runtime(err, "installing helm release %q from chart %q", release.Chart, release.ReleaseName)
+	}
+
+	return rel, nil
+}
+
+func uninstallRelease(
```

Comment:
Nit: this is uninstalling multiple releases.

```suggestion
func uninstallReleases(
```

Reply:
fixed